jujugui/static/gui/src/app/views/ghost-deployer-extension.js

```diff
@@ -127,6 +134,36 @@ YUI.add('ghost-deployer-extension', function(Y) {
     },
 
     /**
+      If we're using Juju 1 then we need to deploy a charm Id which has
+      the series defined in the URL. This is not required for Juju 2 as it
+      supports multi-series charms.
+
+      @method _addSeriesToCharmId
+      @param {String} charmId The charm id.
+      @param {String} series The series of the service.
+    */
+    _addSeriesToCharmId: function(charmId, series) {
+      let charmIdParts = charmId.replace('cs:', '').split('/');
```

Comment:
I would be nice to have a charm/bundle URL parsing library, returning an object with series, name, revision etc attributes, able to be re-serialized as an URL string. We have been having this splitting logic forever everywhere in the GUI. Not in this branch of course.